service/frontend/handler.go

```diff
@@ -391,6 +402,9 @@ func (wh *WorkflowHandler) RecordActivityTaskHeartbeat(
 	sw := wh.startRequestProfile(scope)
 	defer sw.Stop()
 
+	// Count the request in the RPS, but we still accept it even if RPS is exceeded
```

Comment:
Yes. The idea here is that the client cannot generally generate an infinite number of non-throttled API requests without having to go through the throttled ones. e.g. Have to start a workflow or poll for a task in order to be able to call RespondDecisionTaskCompleted, etc.